src/edu/ucsb/nceas/metacat/doi/datacite/EML2DataCiteFactory.java

```diff
@@ -0,0 +1,258 @@
+                    Document doc = generateROOTDoc();
+                    
+                    //identifier
+                    String scheme = DOI;
+                    String id = removeIdSchemePrefix(identifier.getValue(), scheme);
+                    addIdentifier(doc, id, scheme);
+                    
+                    //creator
+                    appendCreators(sysmeta.getRightsHolder(), emlPackage, doc);
+                    
+                    //title
+                    String title = emlPackage.getTitle();
+                    if(title == null || title.trim().equals("")) {
+                        throw new InvalidRequest(INVALIDCODE, "The datacite instance must have a title. It can't be null or blank");
+                    }
+                    appendTitle(title, doc, language);
+                    
+                    //publish
+                    Node node = MNodeService.getInstance(null).getCapabilities();
+                    addPublisher(doc,node.getName());
+
+                    //publication year
+                    SimpleDateFormat sdf = new SimpleDateFormat("yyyy");
+                    String year = sdf.format(sysmeta.getDateUploaded());
+                    addPublicationYear(doc, year);
+                    
+                    //subjects (keywords)
+                    List<String> subjects = emlPackage.getKeywords();
+                    if(subjects != null) {
+                        for(String subject : subjects) {
+                            appendSubject(subject, doc, language);
+                        }
+                    }
+                    
+                    //language
+                    addLanguage(doc, language);
+                    
+                    
+                    //resource type
+                    //String resourceType = lookupResourceType(sysmeta);
+                    String resourceType = null; //only set the attribute to "dataset"
+                    addResourceType(doc, DataCiteProfileResourceTypeValues.DATASET.toString(), resourceType);
+                    
+                    //version
+                    
+                    //description (abstract)
+                    String description = emlPackage.getAbsctrac();
+                    if(description != null) {
+                        appendDescription(description, doc, language, ABSTRACT);
+                    }
+                    //size
+
+                    // format
+                    String format = lookupFormat(sysmeta);
+                    if(format != null) {
+                       appendFormat(doc, format);
+                    }
+                    return serializeDoc(doc);
+                } else {
+                    logMetacat.warn("EML2DataCiteFactory.generateMetadata - the eml package is null and we have to use the default factory");
+                    DefaultDataCiteFactory factory = new DefaultDataCiteFactory();
```

Comment:
See my other comments about falling back to a default factory.  It seems we should be throwing an error since we don't handle the format.  The default factory is adding in some metadata that isn't correct (like `rightsHolder`).